core/trino-main/src/main/java/io/trino/operator/SqlMergeOperator.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.operator;
+
+import io.trino.Session;
+import io.trino.spi.Page;
+import io.trino.spi.connector.ConnectorMergeSink;
+import io.trino.split.PageSinkManager;
+import io.trino.sql.planner.plan.PlanNodeId;
+import io.trino.sql.planner.plan.TableWriterNode.MergeTarget;
+
+import static com.google.common.base.Preconditions.checkState;
+import static io.airlift.concurrent.MoreFutures.toListenableFuture;
+import static java.util.Objects.requireNonNull;
+
+public class SqlMergeOperator
```

Comment:
remove "Sql"

Reply:
> remove "Sql"

Sadly, there is already an unrelated `MergeOperator`.  I suppose another alternative would be to rename that operator.